src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/SnowflakeQuery.scala

```diff
@@ -89,6 +92,18 @@ private[querygeneration] abstract sealed class SnowflakeQuery {
   }
 }
 
+private[querygeneration]
```

Comment:
Introduce `UnarySnowflakeQuery` and `BinarySnowflakeQuery` for code reuse.
Logically, we may introduce `LeafSnowflakeQuery` and `MultipleChildrenSnowflakeQuery`.
But it sounds not necessary because `SourceQuery` and `Union` are the only classes belong to them.
So just enhance  `SourceQuery` and `Union` to implement it.